pkg/ingester/ingester.go

```diff
@@ -150,7 +150,11 @@ type Ingester struct {
 
 	// One queue per flush thread.  Fingerprint is used to
 	// pick a queue.
-	flushQueues []*util.PriorityQueue
+	flushQueues     []*util.PriorityQueue
+	flushQueuesDone sync.WaitGroup
+
+	// Virtual lookup for this function to allow the tests to get in there.
+	flushUserSeries func(userID string, fp model.Fingerprint, immediate bool) error
```

Comment:
I'd rather do this by leaving `flushUserSeries` as it was and instead having a `preFlushUserSeries` hook that is called by `flushUsersSeries`. It costs an extra function call each time, but it makes the structure of this object a bit more obvious.